src/main/java/org/osiam/scim/extension/ExtensionsConfiguration.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (C) 2013-2016 tarent solutions GmbH
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.osiam.scim.extension;
+
+import org.hibernate.validator.constraints.NotEmpty;
+import org.osiam.resources.scim.ExtensionFieldType;
+import org.osiam.storage.ExtensionRepository;
+import org.osiam.storage.entities.ExtensionEntity;
+import org.osiam.storage.entities.ExtensionFieldEntity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toSet;
+
+@Component
+@ConfigurationProperties("osiam.scim")
+public class ExtensionsConfiguration {
+
+    private final ExtensionRepository extensionRepository;
+    private final List<Extension> extensions = new ArrayList<>();
+
+    @Autowired
+    public ExtensionsConfiguration(ExtensionRepository extensionRepository) {
+        this.extensionRepository = extensionRepository;
+    }
+
+    @PostConstruct
+    public void createExtensions() {
+        extensions.stream()
+                .filter(extension -> !extensionRepository.existsByUrnIgnoreCase(extension.getUrn()))
+                .map(Extension::toEntity)
+                .forEach(extensionRepository::saveAndFlush);
+    }
+
+    @Valid
+    public List<Extension> getExtensions() {
```

Comment:
Is public visibility a good idea? We are not handling the `ExtensionConfiguration.Extension` anywhere else then in this class, so maybe we could consider private visibility. Or at least package wide default visibility, to keep the Spock test happy in the IDE, although it shouldn't really care.